detection_rules/utils.py

```diff
@@ -192,3 +192,15 @@ def clear():
 
 def clear_caches():
     _cache.clear()
+
+
+def load_rule_contents(rule_file):
+    """Load a rule file from multiple formats."""
+    try:
```

Comment:
i think we should check the extension first `.ndjson` or `.jsonl`, instead of the try/except handling here

Reply:
also if we load a `.ndjson` file, we should make sure that it's only a single rule, and not a collection